File: Java/LearningTypesOfIntent/app/src/main/java/com/example/learningtypesofintent/MainActivity.java
package com.example.learningtypesofintent;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    ImageView imageView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        imageView = (ImageView) findViewById(R.id.imageView2);
        imageView.setImageResource(R.drawable.logo);

        try {
            imageView.setImageURI((Uri) getIntent().getExtras().get(Intent.EXTRA_STREAM));
        } catch (Exception e) {
            Toast.makeText(getApplicationContext(), e.getMessage(), Toast.LENGTH_LONG).show();
        }


    }
}

File: LearningTimer/app/src/main/java/com/example/learningtimer/MainActivity.kt
package com.example.learningtimer

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.CountDownTimer
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val timer = object: CountDownTimer(10000, 1000) {
            override fun onTick(millisUntilFinished: Long) {
                textViewTimer.setText("Remaining time : ${millisUntilFinished/1000}")
            }

            override fun onFinish() {
                Toast.makeText(applicationContext,"Time is up",Toast.LENGTH_LONG).show()
            }
        }
        timer.start()
    }
}